shairport.c

```diff
@@ -586,6 +586,25 @@ int parse_options(int argc, char **argv) {
           config.udp_port_range = value;
       }
 
+      /* Get the TCP port base setting. */
+      if (config_lookup_int(config.cfg, "general.tcp_port_base", &value)) {
+        if ((value < 0) || (value > 65535))
+          die("Invalid port number  \"%sd\". It should be between 0 and 65535, default is 6001",
+              value);
+        else
+          config.tcp_port_base = value;
+      }
+
+      /* Get the TCP port range setting. This is number of ports that will be tried for free ports ,
+       * starting at the port base. */
+      if (config_lookup_int(config.cfg, "general.tcp_port_range", &value)) {
+        if ((value < 3) || (value > 65535))
+          die("Invalid port range  \"%sd\". It should be between 3 and 65535, default is 10",
```

Comment:
Not sure how many ports are needed for AP2. Max I saw during testing was 2, but could have higher minimum to be conservative?